src/fstests/fstests.jinja2

```diff
@@ -46,6 +54,38 @@ class Job(BaseJob):
             print('Problem found during tests:', e)
             return 'fail'
 
+    def _get_xml_from_vm(self):
+        print('calling VM')
+        try:
+            os.chdir(VM_ROOT)
+            result = subprocess.run([f"debugfs -R \"dump ext4/results-4k/results.xml {FSTESTS_ROOT}/results.xml\" {VM_ROOT}/run-fstests/disks/vdg"], check=True, shell=True)
+            if result.returncode == 0:
+                print(f"XML file in the output directory.")
+                return 'pass'
+        except Exception as e:
+            print('XML not found.', e)
+            return 'fail'
+
+    def _parse_fstests_result(self):
+        tree = ET.parse(f"{FSTESTS_ROOT}/results.xml")
+        root = tree.getroot()
+        results = []
+        for child in root.iter('testcase'):
+            name = child.attrib['name']
+            testcase = {}
+            testcase['name'] = name.replace('/', '.')
+            if child.find('skipped') != None:
+                testcase['result'] = None
+            elif child.find('failure') != None:
+                testcase['result'] = 'fail'
+            else:
+                testcase['result'] = 'pass'
+            results.append(testcase)
```

Comment:
There are a few issues here, it's not creating sub-groups and the test names shouldn't have a `.` in them.  Instead, the name should be split to have the path elements.

We can discuss this in more detail outside of GitHub.